### PR TITLE
Add -Wmissing-braces to linux test

### PR DESCRIPTION
**Pull request by peterrum**
These types of warnings, we currently only see on `CDash`. The tests will hopefully fail; this should be fixed once https://github.com/dealii/dealii/pull/10742 is merged.

### CODE REVIEW

**Comment by tjhei** [comment]
Interesting, we should consider turning this on by default. @tamiko ?

**Comment by tjhei** [comment]
It seems to fail as expected:
```
/home/runner/work/dealii/dealii/include/deal.II/grid/reference_cell.h:339:37: error: missing braces around initializer for ‘std::__array_traits<unsigned int, 2>::_Type {aka unsigned int [2]}’ [-Werror=missing-braces]
             {{0, 0}, {0, 1}, {1, 1}}};
```
Can you rebase to  #10742? (Or I guess we can wait until the tests pass)

**Comment by peterrum** [comment]
Done!

**Comment by masterleinad** [comment]
> Interesting, we should consider turning this on by default. @tamiko ?

Yes, I would just add it since we care.

File: include/deal.II/grid/connectivity.h [hunk 1]
```diff
@@ -166,7 +166,7 @@ namespace internal
       unsigned int
       n_entities(const unsigned int d) const override
       {
-        static std::array<unsigned int, 3> table = {2, 1};
+        static std::array<unsigned int, 3> table = {{2, 1}};
```
Comment (masterleinad) [inline include/deal.II/grid/connectivity.h:169]: Unrelated to this pull request, I find omitting the last zero here (and in similar places) rather confusing.

Comment (peterrum) [reply include/deal.II/grid/connectivity.h:169]: This was a mistake. The vector should actually have the length 2. I have fixed it in #10742.

**Pull request merged**